src/main/java/Duke.java

```diff
@@ -1,10 +1,378 @@
+import java.io.FileWriter;
+import java.util.Objects;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
+
+/**
+ * This represents the Duke Chat bot.
+ */
 public class Duke {
+
+    private static boolean isDukeDone = false;
+    private static ArrayList<Task> list = new ArrayList<>();
+
+    private final static String LINES = "    ____________________________________________________________";
+    private final static String TAB = "    ";
+    private final static String GREETING = TAB + "Hello... I'm Sadge Duke\n    What can I do for you? :(";
+    private final static String GOODBYE = TAB + "Please don't go... I'll miss you...";
+    private final static String INVALID_GENERAL = TAB + "I'm really sorry... This is an invalid input...";
+    private final static String INVALID_DONE = TAB + "I'm sorry... This is an incorrect done input...\n"
+            + TAB + "Please use 'list' to see what number can be used...";
+    private final static String INVALID_DELETE = TAB + "Why would you delete a task... Your delete statement is an "
+            + "incorrect input too...";
+    private final static String INVALID_TASK = "Your task is really weird... I don't think I like it...";
+    private final static String TODO = "todo";
+    private final static String DEADLINE = "deadline";
+    private final static String EVENT = "event";
+    private final static String FILE_PATH = "./data/tasks.txt";
+
+    private final static int DESCRIPTION = 0;
+    private final static int DATETIME = 1;
+    private final static int TASK_TYPE_INDEX = 0;
+    private final static int IS_DONE_INDEX = 1;
+    private final static int TASK_INDEX = 2;
+    private final static int BY_AT_INDEX = 3;
```

Comment:
Would it be better if you group similar constants together with the same name format such as `MESSAGE_GREETING` and `MESSAGE_GOODBYE`? Also, you are recommended to place the same type at the front for the naming. For example, `INDEX_IS_DONE` and `INDEX_TASK`. It allows you to trace through code easier.